File: app/src/main/java/com/kelsos/mbrc/connection_manager/ConnectionManagerPresenter.java
package com.kelsos.mbrc.connection_manager;

import com.kelsos.mbrc.data.ConnectionSettings;
import com.kelsos.mbrc.presenters.Presenter;

public interface ConnectionManagerPresenter extends Presenter<ConnectionManagerView> {
  void load();

  void setDefault(ConnectionSettings settings);

  void update(ConnectionSettings settings);

  void delete(ConnectionSettings settings);
}

File: app/src/main/kotlin/com/kelsos/mbrc/connection_manager/ConnectionManagerPresenter.kt
package com.kelsos.mbrc.connection_manager

import com.kelsos.mbrc.data.ConnectionSettings
import com.kelsos.mbrc.presenters.Presenter

interface ConnectionManagerPresenter : Presenter<ConnectionManagerView> {
  fun load()

  fun setDefault(settings: ConnectionSettings)

  fun save(settings: ConnectionSettings)

  fun delete(settings: ConnectionSettings)
}
